lib/ruby_lsp/ruby_lsp_rails/server.rb

```diff
@@ -529,6 +529,22 @@ def database_supports_indexing?(model)
   end
 end
 
+# Patch fork to name processes based on the caller's file path. This is useful for figuring out what is creating more
+# child processes from the runtime server, so that we can optimize and more easily debug orphaned processes
+# @requires_ancestor: Kernel
+module ForkHook
+  #: (*untyped) -> Integer?
+  def _fork(*args)
+    fork_caller = caller_locations(1, 1)&.first
+
+    pid = super
+    Process.setproctitle("ruby-lsp-rails: #{fork_caller.path}") if fork_caller && pid == 0
+    pid
```

Comment:
Small optimization here:
```suggestion
    pid = super
    
    if pid == 0
      fork_caller = caller_locations(1, 1)&.first

      Process.setproctitle("ruby-lsp-rails: #{fork_caller.path}") if fork_caller
    end
    
    pid
```

so that we only call `caller_locations` if we wanted to set the process title.